src/components/MoveReference/index.tsx

```diff
@@ -1,241 +1,221 @@
-import React, { useEffect, useState } from "react";
+import React, { useEffect, useRef, useState } from "react";
 import BrowserOnly from "@docusaurus/BrowserOnly";
 import ReactMarkdown from "react-markdown";
 import rehypeRaw from "rehype-raw";
 import remarkGfm from "remark-gfm";
-
 import ExecutionEnvironment from "@docusaurus/ExecutionEnvironment";
+import { Select, Label, Button, ListBox, ListBoxItem, Popover, SelectValue, ComboBox, Input, Section, Header } from 'react-aria-components';
+import { IconChevronDown } from '@tabler/icons-react';
 
-const root = "https://raw.githubusercontent.com/aptos-labs/aptos-core/";
-const url_root = "/reference/move";
-
+const root = "https://raw.githubusercontent.com/aptos-labs/aptos-core";
 const branches = ["mainnet", "testnet", "devnet", "main"];
-
-const branch_titles = ["Mainnet", "Testnet", "Devnet", "Main"];
-
-const frameworks = [
-  "move-stdlib",
-  "aptos-stdlib",
-  "aptos-framework",
-  "aptos-token",
-  "aptos-token-objects",
-];
-const TopNav = ({ branch }: TopNavProps) => {
-  const adjustBranch = (event) => {
-    const params = new URLSearchParams(window.location.search);
-    params.set("branch", event.target.getAttribute("branch"));
-    window.location.href = `${location.pathname}?${params.toString()}`;
+const branchTitles = ["Mainnet", "Testnet", "Devnet", "Main"];
+const frameworks = ["move-stdlib", "aptos-stdlib", "aptos-framework", "aptos-token", "aptos-token-objects"];
+const defaultFramework = "move-stdlib";
+
+const TopNav = ({ branch, onBranchChange }) => (
+  <Select selectedKey={branch} onSelectionChange={onBranchChange}>
+    <Label>Branch</Label>
+    <Button>
+      <SelectValue>{branchTitles[branches.indexOf(branch)]}</SelectValue>
+      <IconChevronDown aria-hidden="true" />
+    </Button>
+    <Popover>
+      <ListBox>
+        {branches.map((b, index) => (
+          <ListBoxItem key={b} id={b}>{branchTitles[index]}</ListBoxItem>
+        ))}
+      </ListBox>
+    </Popover>
+  </Select>
+);
+
+const FrameworkSelector = ({ frameworkData, selectedFramework, onSelectFramework }) => {
+  const itemRefs = useRef({});
+  const [isComboBoxOpen, setOpen] = useState(false);
+
+  // Handler for framework selection changes
+  const handleSelectionChange = (selectedFrameworkUrl) => {
+    onSelectFramework(selectedFrameworkUrl);
```

Comment:
I think in your demo video.  `Framework` is actually `Module`, I think you're asking users to select a module.

I would cleanup the following:
1. `move-stdlib`, `aptos-stdlib` etc. are `Packages`
2. `0x1::string`, `0x1::account` are `Modules`